faker/providers/person/th_TH/__init__.py

```diff
@@ -62,29 +68,87 @@ class Provider(PersonProvider):
-        "วศิน", "อภิสรา", "ณัฐวรรณ", "ธันรดี", "นิรชา", "ธันยชนก", "กมลชนก", "จารุวรรณ",
-        "ผจงรักษ์", "ภัคจิรา", "ปัทธมพร", "สุวิจักษ์", "นพเก้า", "รัชชานนท์", "อธิศ",
-        "คุณาพร", "อริสรา", "ทัศวรรณ", "ชัยวัฒน์", "ภัณฑิรา", "ศุภสุตา", "พัชรพร",
-        "ภูเบศ",
-        "พัทธมน", "ชัญญา", "ปัณณวิชย์", "ชวิน", "ภาดา", "ชนิกานต์", "ณัฐวดี", "ชาลิสา",
-        "ประเมศฐ์", "ศุภสิทธิ์", "สิทธิวัต", "ธีธีช", "ญาดา", "เพ็ญพิชชา", "อนนต์",
-        "ชายฟ้า",
-        "ปวัน", "อัญญาร", "ญานิศา", "ภาคย์", "ชญานินท์", "ชยาภัทร", "จิตริน", "วัศยา",
-        "ปิติภัทร", "นิชากานต์", "ปริน", "ธนัชชา",)
+    # female first name
+    # http: // www.mamaexpert.com / posts / content - 1166
+    first_names_female = (
+        "กัสจัง", "เกรซ", "เกรส", "แก้ม", "ข้าวสวย", "ข้าวหอม", "ข้าวทิพย์", "ข้าวปุ้น",
+        "แจน", "ณิชา", "นินิว", "นานา", "นิว", "นับตังค์", "เนเน่", "น้ำมนต์", "โบวี่",
+        "โบนัส", "บิวตี้", "ใบเตย", "ใบบัว", "ใบข้าว", "ใบตอง", "ใบเฟิร์น", "พรีช",
+        "พอใจ",
+        "พอเพียง", "ฟ้าใส", "เฟิร์ส", "เฟสบุค", "เฟสบุ๊ค", "ต้นข้าว", "ต้นหอม", "แตงกว่า",
+        "แตงโม", "ตังค์", "ตังค์", "มิวสิค", "มะนาว", "มิ๊น", "มิ๊นท์", "มินนี่", "บัว",
+        "ปันปัน", "แพท", "แพต", "แพทตี้", "เเพทตี้", "ลูกพีช", "ภูมิใจ", "เอิร์น", "เอิน",
+        "เอวา", "เอมิ", "เอมี่", "ออมสิน", "ออกัส", "ไอซ์",)
+
+    # male first name
+    ##http: // www.mamaexpert.com / posts / content - 1166
+    first_names_male = (
+        "กันต์", "กันย์", "กาย", "กัปตัน", "กูเกิ้ล", "ขุนพล", "ข้าวปั้น", "คุณ",
+        "เจ้าคุณ",
+        "เจ้าขุน", "เจ้านาย", "ช็อปเปอร์", "โชกุน", "ชิณณ์", "ชิน", "ตุล", "ตุลย์",
+        "โตโน่",
+        "ตะวัน", "ไตตั้น", "ไตเติ้ล", "ปกป้อง", "ต้นน้ำ", "ต้นกล้า", "นะโม", "ไบรท์",
+        "เบส",
+        "ปลื้ม", "ปุณ", "ปุณณ์", "ปุญญ์", "แทนคุณ", "ธัน", "ธันวา", "พอร์ช", "พัตเตอร์",
+        "พอร์ตเตอร์", "พอตเตอร์", "ฟลุ๊ก", "ฟลุ๊ค", "เฟสบุ๊ค", "ภาคิน", "ภูมิ", "ภูผา",
+        "ภาคภูมิ", "ภู", "ภูมิใจ", "มาวิน", "มีคุณ", "ออโต้", "ออดี้", "อิ๊คิว", "อิคคิว",
+        "อิคคิวซัง", "อิกคิว", "อะตอม", "ไอซ์", "อั่งเปา", "ออกัส", "อชิ", "อะชิ",
+        "เอิร์ธ",
+        "กันต์", "กันย์", "กาย", "กัปตัน", "กูเกิ้ล", "ขุนพล", "ข้าวปั้น", "คุณ",
+        "เจ้าคุณ",
+        "เจ้าขุน", "เจ้านาย", "ช็อปเปอร์", "โชกุน", "ชิณณ์", "ชิน", "ตุล", "ตุลย์",
+        "โตโน่",
+        "ตะวัน", "ไตตั้น", "ไตเติ้ล", "ปกป้อง", "ต้นน้ำ", "ต้นกล้า", "นะโม", "ไบรท์",
+        "เบส",
+        "ปลื้ม", "ปุณ", "ปุณณ์", "ปุญญ์", "แทนคุณ", "ธัน", "ธันวา", "พอร์ช", "พัตเตอร์",
+        "พอร์ตเตอร์", "พอตเตอร์", "ฟลุ๊ก", "ฟลุ๊ค", "เฟสบุ๊ค", "ภาคิน", "ภูมิ", "ภูผา",
+        "ภาคภูมิ", "ภู", "ภูมิใจ", "มาวิน", "มีคุณ", "ออโต้", "ออดี้", "อิ๊คิว", "อิคคิว",
+        "อิคคิวซัง", "อิกคิว", "อะตอม", "ไอซ์", "อั่งเปา", "ออกัส", "อชิ", "อะชิ",
+        "เอิร์ธ",)
+
+    first_names = first_names_female + first_names_male
+
+    # last name
+    # http://www.reurnthai.com/index.php?topic=5273.45
```

Comment:
I'm not sure what this url is. Seems to be a forum thread about literature. I think a comment explaining why you're picking names from the there would be helpful, for people not familiar with Thai culture.